angular/openmrs-rest/openmrs-rest.js

```diff
@@ -130,6 +139,8 @@ function openmrsRest() {
 			query = addMode(query, 'ref');
 			return openmrsApi[resource].get(query).$promise.then(function (response) {
 				return new PartialList(response, $document);
+			}, function(error){
+				return redirectToLogin(error)
```

Comment:
You should handle the error not only for list operations but all inc. retire, update, etc.